tasks/ModelTask.cpp

```diff
@@ -30,101 +29,102 @@ void ModelTask::setGazeboModel(WorldPtr _world,  ModelPtr _model)
 
 	world = _world;
 	model = _model;
-//	sdf = model->GetSDF();
 	
-//	Export Gazebo Joints and Links to Rock-Robotics
 	setJointPorts();
-	setLinkPorts();
 } 
-//======================================================================================
+
 void ModelTask::setJointPorts()
 {
-    // base::samples::Joints rock_joint; 
-	// Get all joints from a model and creates a Rock component InputPort
-	joints = model->GetJoints();
-	for(Joint_V::iterator joint = joints.begin(); joint != joints.end(); ++joint)
+	// Get all joints from a model and set Rock Input/Output Ports
+	gazebo_joints = model->GetJoints();
+	for(Joint_V::iterator joint = gazebo_joints.begin(); joint != gazebo_joints.end(); ++joint)
 	{
-		gzmsg << "RockBridge: found joint: " << world->GetName() + "/" + model->GetName() + 
+		gzmsg << "RockBridge: found joint: " << world->GetName() + "/" + model->GetName() +
 				"/" + (*joint)->GetName() << std::endl;
-		gzmsg << "RockBridge: create joint InputPort in Rock." << std::endl;
-		joint_port = new RTT::InputPort<double>( "_joint_" + (*joint)->GetName() + "_in");
-		ports()->addPort( *joint_port );
-		joint_port_list.push_back( std::make_pair(joint_port,*joint) );			
+
+        joints_in.names.push_back( (*joint)->GetName() );
+        joints_in.elements.push_back( base::JointState::Effort(0.0) );
+        joints_out.names.push_back( (*joint)->GetName() );
+        joints_out.elements.push_back( base::JointState::Position(0.0) );
 	}
 }
-//======================================================================================
+
 void ModelTask::setLinkPorts()
 {
-    // Get all links from a model and creates a Rock component InputPort
-	links = model->GetLinks();
-	for(Link_V::iterator link = links.begin(); link != links.end(); ++link)
+    // Create Rock output ports for the links defined in the configuration file.
+	gazebo_links = model->GetLinks();
+	for(Link_V::iterator link = gazebo_links.begin(); link != gazebo_links.end(); ++link)
 	{
-		gzmsg << "RockBridge: found link: " << world->GetName() + "/" + model->GetName() + 
-				"/" + (*link)->GetName() << std::endl;
-		gzmsg << "RockBridge: create link InputPort in Rock." << std::endl;
-		link_port = new RTT::InputPort<base::Vector3d>( "_link_" + (*link)->GetName() + "_in");
-		ports()->addPort( *link_port );
-		link_port_list.push_back( std::make_pair(link_port,*link) );
+        for(NameVector::iterator it = link_names.begin(); it != link_names.end(); ++it)
+        {
+            if( (*it).compare( (*link)->GetName() ) == 0 )
```

Comment:
It is not natural to use .compare() to compare strings for *equality* in C++. compare() is used for ordering (returns -1, 0 or 1). Use str1 == str2 to compare for equality.